lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -210,8 +210,7 @@ def doom_reason
         elsif limbo_reason == "CrashLoopBackOff"
           exit_code = @status.dig('lastState', 'terminated', 'exitCode')
           "Crashing repeatedly (exit #{exit_code}). See logs for more information."
-        elsif %w(ImagePullBackOff ErrImagePull).include?(limbo_reason) &&
-          limbo_message.match(/(?:not found)|(?:back-off)/i)
+        elsif %w(ErrImagePull ImagePullBackOff).include?(limbo_reason) && limbo_message.match(/not found/i)
```

Comment:
Alternatively, we could remove this entirely. In practice I don't think `ImagePullBackoff` ever includes "not found" in existing k8s versions, so this detection is extremely racy. I can see the argument that a consistent timeout would be better for UX.

Reply:
I think we should try and be helpful where we can.

Reply:
>  In practice I don't think ImagePullBackoff ever includes "not found" in existing k8s versions

This comes from here I think: https://github.com/kubernetes/kubernetes/blob/d24fe8a801748953a5c34fd34faa8005c6ad1770/pkg/kubelet/images/image_manager.go#L126

```
fmt.Sprintf("Back-off pulling image %q", container.Image)
```

Shouldn't we then remove `ImagePullBackOff` because it would always fail the `not found` string match?

Reply:
Yes, it was lazy of me not to look up the source. Thanks for finding it! Since it is confirmed to not be possible for the error to match, I agree that keeping ImagePullBackoff here is just misleading.